ddl/reorg.go

```diff
@@ -134,6 +136,44 @@ func (rc *reorgCtx) getRowCount() int64 {
 	return row
 }
 
+func getDistAddIndexRowCntAndSetJobRowCnt(ctx context.Context, reorgInfo *reorgInfo, rc *reorgCtx, job *model.Job, client *clientv3.Client) int64 {
+	rowCount := int64(0)
+	if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
+		path := fmt.Sprintf("distAddIndex/%d", job.ID)
+		resp, err := client.Get(ctx, path, clientv3.WithPrefix())
+		if err != nil {
+			logutil.BgLogger().Warn("[ddl] get row count from ETCD failed", zap.Error(err))
+			return 0
+		}
+		if len(resp.Kvs) == 0 {
+			return 0
+		}
+		for _, kv := range resp.Kvs {
+			cnt, err := strconv.Atoi(string(kv.Value))
+			if err != nil {
+				logutil.BgLogger().Error("[ddl] parse row count from ETCD failed", zap.Error(err))
+				continue
+			}
+			rowCount += int64(cnt)
+		}
+		job.SetRowCount(rowCount)
+	} else {
+		rowCount = rc.getRowCount()
+		job.SetRowCount(rowCount)
+	}
+	return rowCount
+}
+
+func gcETCDRowCntStatIfNecessary(ctx context.Context, reorgInfo *reorgInfo, job *model.Job, client *clientv3.Client) {
+	if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
+		path := fmt.Sprintf("distAddIndex/%d", job.ID)
+		_, err := client.Delete(ctx, path, clientv3.WithPrefix())
+		if err != nil {
+			logutil.BgLogger().Warn("[ddl] delete row count from ETCD failed", zap.Error(err))
+		}
+	}
+}
```

Comment:
The function name is nothing related to the body. It should be `cleanDistributedRowCount()`.